ctapipe/io/dl1eventsource.py

```diff
@@ -0,0 +1,389 @@
+from ctapipe.io.datalevels import DataLevel
+from ctapipe.instrument import (
+    TelescopeDescription,
+    SubarrayDescription,
+    CameraGeometry,
+    OpticsDescription,
+)
+from ctapipe.containers import EventAndMonDataContainer, EventType
+from ctapipe.containers import MCHeaderContainer, MCEventContainer
+from ctapipe.io.eventsource import EventSource
+from astropy.coordinates import Angle
+import astropy.units as u
+import tables
+import warnings
+
+
+class DL1EventSource(EventSource):
+    def __init__(
+        self,
+        input_url,
+        config=None,
+        parent=None,
+        **kwargs
+    ):
+        """
+        EventSource for dl1 files in the standard DL1 data format
+
+        Parameters:
+        -----------
+        input_url : str
+            Path of the file to load
+        config: ??
+        parent: ??
+        kwargs
+        """
+        super().__init__(
+            input_url=input_url,
+            config=config,
+            parent=parent,
+            **kwargs
+        )
+
+        self.file_ = tables.open_file(input_url)
+        self._subarray_info = self._prepare_subarray_info(
+            self.file_.root.configuration.instrument
+        )
+        self._mc_header = self._parse_mc_header(
+            self.file_.root.configuration.simulation.run
+        )
+
+    def __exit__(self, exc_type, exc_val, exc_tb):
+        self.close()
+
+    def close(self):
+        self.file_.close()
+
+    @staticmethod
+    def is_compatible(file_path):
+        """
+        Implementation needed!
+        """
+        return False
+
+    @property
+    def is_simulation(self):
+        """
+        Implementation needed!
+        """
+        return True
+
+    @property
+    def subarray(self):
+        return self._subarray_info
+
+    @property
+    def datalevels(self):
+        return (DataLevel.DL1_IMAGES, DataLevel.DL1_PARAMETERS)
+
+    @property
+    def obs_id(self):
+        return set(self.file_.root.dl1.event.subarray.trigger.col("obs_id"))
+
+    @property
+    def mc_header(self):
+        return self._mc_header
+
+    def _generator(self):
+        try:
+            yield from self._generate_events()
+        except EOFError:
```

Comment:
This should not really happen with hdf5 files, that's copy paste from the simtel source I assume